.github/workflows/main.yml

```diff
@@ -1,61 +1,107 @@
+  #     steps:
+  #       - uses: actions/checkout@v2
+  #       - name: Docker login
+  #         run: docker login docker.pkg.github.com -u $GITHUB_ACTOR -p $GITHUB_TOKEN
+  #         env:
+  #           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+  #       - name: Build in container
+  #         env:
+  #           CI_NETWORK: true
+  #           CI: true
+  #         run: |
+  #           echo $GITHUB_WORKSPACE
+  #           docker run --privileged -e CI=true -t -v $GITHUB_WORKSPACE:/github/workspace docker.pkg.github.com/fwupd/fwupd/fwupd-${{matrix.os}}:latest
+  # 
+  #   fuzzing:
+  #     runs-on: ubuntu-latest
+  #     steps:
+  #     - name: Build Fuzzers
+  #       id: build
+  #       uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
+  #       with:
+  #         oss-fuzz-project-name: 'fwupd'
+  #         dry-run: false
+  #     - name: Run Fuzzers
+  #       uses: google/oss-fuzz/infra/cifuzz/actions/run_fuzzers@master
+  #       with:
+  #         oss-fuzz-project-name: 'fwupd'
+  #         fuzz-seconds: 300
+  #         dry-run: false
+  #     - name: Upload Crash
+  #       uses: actions/upload-artifact@v1
+  #       if: failure() && steps.build.outcome == 'success'
+  #       with:
+  #         name: artifacts
+  #         path: ./out/artifacts
 
-  build:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        os: [fedora, debian-x86_64, arch, debian-i386]
+  build-freebsd:
+    runs-on: macos-latest
+    name: Build fwupd from FreeBSD ports
     steps:
-      - uses: actions/checkout@v2
-      - name: Docker login
-        run: docker login docker.pkg.github.com -u $GITHUB_ACTOR -p $GITHUB_TOKEN
-        env:
-          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
-      - name: Build in container
-        env:
-          CI_NETWORK: true
-          CI: true
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Find tag
+      id: tagger
+      uses: jimschubert/query-tag-action@v1
+    - name: Build
+      id: test
+      uses: vmactions/freebsd-vm@v0.1.2
+      with:
+        usesh: true
+        mem: 8192
+        prepare: |
+          pkg install -y git python3 glib meson pkgconf gobject-introspection \
+            vala gtk-doc json-glib gpgme gnutls sqlite3 curl gcab libarchive
+        sync: rsync
         run: |
-          echo $GITHUB_WORKSPACE
-          docker run --privileged -e CI=true -t -v $GITHUB_WORKSPACE:/github/workspace docker.pkg.github.com/fwupd/fwupd/fwupd-${{matrix.os}}:latest
+          mkdir -p /usr/local/etc/pkg/repos/
+          cp /etc/pkg/FreeBSD.conf /usr/local/etc/pkg/repos/FreeBSD.conf
+          sed -i .old 's|url: "pkg+http://pkg.FreeBSD.org/${ABI}/quarterly"|url: "pkg+http://pkg.FreeBSD.org/${ABI}/latest"|' /usr/local/etc/pkg/repos/FreeBSD.conf
+          pkg install -y libtasn1
+          pkg upgrade -y libtasn1
+          cd /usr
+          git clone https://github.com/3mdeb/freebsd-ports.git --depth 1 -b fwupd-port ports
```

Comment:
@superm1 @hughsie @SergiiDmytruk We should push this forward in the near future. More and more FreeBSD-related changes are coming to fwupd, so a proper CI would be really useful. The upstream of the [fwupd port](https://reviews.freebsd.org/D29332) will take some more time, so at the start we should base CI on the [freebsd-ports fork](https://github.com/3mdeb/freebsd-ports.git), which will contain the fwupd package. 
The FreeBSD related patches are already merged in the fwupd, so this PR needs significant changes. I will address this in the coming days.

Reply:
@Asiderr can you create a wip/freebsd branch on the upstream fwupd repo please, and then we can all just commit fixes on before we squash everything up for commit. If you just **add** the freesbd CI job there we can see what's left to upstream.